### PR TITLE
e2e: compact with latest rev in alarm test

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/6677.

I can reproduce the failure pretty easily.

It fails when the available quota is X, and
the compact rev is not recent enough to make
free space.

It can happen because we set the db size periodically
whenever commit happens in mvcc side. Quota layer's
db size might be stale if the commit hasn't happened
recently.

@xiang90

### CODE REVIEW

**Comment by gyuho** [comment]
Example log

```
compacting 6
2016-10-19 15:55:02.045473 I | mvcc/backend: commit starts
2016-10-19 15:55:02.052942 I | mvcc/backend: commit with pending 4 , size: 90112
2016-10-19 15:55:02.052955 I | mvcc: store.index: compact 6
2016-10-19 15:55:02.053014 I | mvcc: finished scheduled compaction at 6 (took 25.794µs)
2016-10-19 15:55:02.053027 I | mvcc/backend: commit starts
2016-10-19 15:55:02.056520 I | mvcc/backend: commit with pending 4 , size: 90112
compacted revision 6
compacted 6
calling defrag
2016-10-19 15:55:02.071121 N | etcdserver/api/v3rpc: starting to defragment the storage backend...
2016-10-19 15:55:02.071143 I | mvcc/backend: commit starts
2016-10-19 15:55:02.095759 I | mvcc/backend: commit starts
2016-10-19 15:55:02.095785 I | mvcc/backend: commit with pending 0, size 53248
2016-10-19 15:55:02.095793 N | etcdserver/api/v3rpc: finished defragmenting the storage backend
Finished defragmenting etcd member[localhost:20000]
after defrag
2016-10-19 15:55:02.097402 I | mvcc/backend: commit starts
2016-10-19 15:55:02.097425 I | mvcc/backend: commit with pending 0, size 53248
2016-10-19 15:55:02.117844 I | etcdserver: alarm disarmed action:DEACTIVATE memberID:47555275972384877 alarm:NOSPACE 
memberID:47555275972384877 alarm:NOSPACE 
calling ctlV3Put
available: 53248 328 53248
```

Quota size is `53248` and the db size after compact with stale revision + defrag is also `53248`

File: e2e/ctl_v3_alarm_test.go [hunk 1]
```diff
@@ -57,7 +57,8 @@ func alarmTest(cx ctlCtx) {
 	}
 
 	// make some space
-	if err := ctlV3Compact(cx, rev, true); err != nil {
+	// compact with the latest revision (store.currentRev.main)
+	if err := ctlV3Compact(cx, rev+2, true); err != nil {
```
Comment (xiang90) [inline e2e/ctl_v3_alarm_test.go]: i do not understand this. we want to compact the keys put before line 47. they should occupy the majority of space. we need to figure out why compact them wont release space.

Comment (gyuho) [inline e2e/ctl_v3_alarm_test.go]: Ok I will investigate more on compaction side. Thanks.

**Comment by gyuho** [comment]
@xiang90 PTAL. Now gets latest revision with a separate API call. And compact on it.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**